pkg/api/v1/orchestrator/routes/handlers.go

```diff
@@ -92,34 +92,28 @@ func (r *Routes) conditionStatusUpdate(c *gin.Context) (int, *v1types.ServerResp
 	}
 
 	// expect condition to be in an incomplete state to accept this update.
-	cr, err := r.repository.Get(ctx, serverID)
+	cond, err := r.repository.GetActiveCondition(ctx, serverID)
 	if err != nil {
-		return http.StatusInternalServerError, &v1types.ServerResponse{
-			Message: "condition lookup: " + err.Error(),
+		if errors.Is(err, store.ErrConditionNotFound) {
+			return http.StatusBadRequest, &v1types.ServerResponse{
```

Comment:
Arguably should be `StatusNotFound`.

Reply:
thanks, I was conflicted between it being a incorrect request vs a 404. Updated all the handlers to return 404s.